card/src/main/java/com/adyen/checkout/card/internal/ui/view/StoredCardView.kt

```diff
@@ -0,0 +1,173 @@
+            // Prevent taking screenshot and screen on recents.
+            getActivity(context)?.window?.addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        if (!BuildUtils.isDebugBuild(context)) {
+            getActivity(context)?.window?.clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        }
+    }
+
+    override fun initView(delegate: ComponentDelegate, coroutineScope: CoroutineScope, localizedContext: Context) {
+        if (delegate !is CardDelegate) throw IllegalArgumentException("Unsupported delegate type")
+        cardDelegate = delegate
+
+        this.localizedContext = localizedContext
+        initLocalizedStrings(localizedContext)
+
+        observeDelegate(delegate, coroutineScope)
+        initSecurityCodeInput()
+    }
+
+    private fun initLocalizedStrings(localizedContext: Context) {
+        binding.textInputLayoutCardNumber.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_CardNumberInput,
+            localizedContext
+        )
+        binding.textInputLayoutExpiryDate.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_ExpiryDateInput,
+            localizedContext
+        )
+        binding.textInputLayoutSecurityCode.setLocalizedHintFromStyle(
+            R.style.AdyenCheckout_Card_SecurityCodeInput,
+            localizedContext
+        )
+    }
+
+    private fun observeDelegate(delegate: CardDelegate, coroutineScope: CoroutineScope) {
+        delegate.outputDataFlow
+            .onEach { outputDataChanged(it) }
+            .launchIn(coroutineScope)
+    }
+
+    private fun initSecurityCodeInput() {
+        val securityCodeEditText = binding.textInputLayoutSecurityCode.editText as? SecurityCodeInput
+        securityCodeEditText?.setOnChangeListener { editable: Editable ->
+            cardDelegate.updateInputData { securityCode = editable.toString() }
+            binding.textInputLayoutSecurityCode.hideError()
+        }
+        securityCodeEditText?.onFocusChangeListener = OnFocusChangeListener { _, hasFocus ->
+            val securityCodeValidation = cardDelegate.outputData.securityCodeState.validation
+            if (hasFocus) {
+                binding.textInputLayoutSecurityCode.hideError()
+            } else if (securityCodeValidation is Validation.Invalid) {
+                binding.textInputLayoutSecurityCode.showError(localizedContext.getString(securityCodeValidation.reason))
+            }
+        }
+    }
+
+    private fun outputDataChanged(cardOutputData: CardOutputData) {
+        binding.editTextCardNumber.setText(
+            localizedContext.getString(
+                R.string.card_number_4digit,
+                cardOutputData.cardNumberState.value
+            )
+        )
+        binding.editTextExpiryDate.date = cardOutputData.expiryDateState.value
+        binding.textInputLayoutSecurityCode.takeIf { isVisible }?.requestFocus()
```

Comment:
Why are we requesting focus when the output data changes?

Reply:
I should move it to `initSecurityCodeInput`.